src/main/java/controller/GameController.java

```diff
@@ -0,0 +1,119 @@
+package controller;
+
+import view.InputView;
+import view.OutputView;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class GameController {
+    private static final int ADVANCE_DETERMINATION_RANGE = 10;
+    private static final String DEFAULT_DELIMITER = ",";
+    private static final int MAX_NAME_LENGTH = 5;
+    private static final Random random = new Random();
+    private static final int START_POSITION = 0;
+    private static final int ADVANCE_CONDITION = 4;
+    private static final int ADVANCE = 1;
+    private static final int GAME_OVER = 0;
+
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public GameController(final InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public void start() {
+        String names = inputView.inputNames();
+        int rounds = inputView.inputRounds();
+        List<String> carNames = split(names);
+        validateNameLength(carNames);
+        Map<String, Integer> cars = createCars(carNames);
+        outputView.printResult();
+        race(rounds, cars);
+        List<String> winners = announceWinners(cars);
+        outputView.printWinners(winners);
+    }
+
+    public List<String> split(String names) {
+        validateDelimiter(names);
+        return Arrays.stream(names.split(DEFAULT_DELIMITER))
+                .map(String::trim)
+                .collect(Collectors.toList());
+    }
+
+    private void validateDelimiter(String names) {
+        if (!names.contains(DEFAULT_DELIMITER)) {
+            throw new RuntimeException("\",\"가 포함되어야 합니다.");
+        }
+    }
+
+    public void validateNameLength(List<String> names) {
+        boolean isOverLength = names.stream()
+                .anyMatch(name -> name.length() > MAX_NAME_LENGTH);
```

Comment:
anyMatch 사용 👍

Reply:
혹은 아래와 같은 사용도 가능합니다.
예외처리까지 stream 체이닝 안 쪽으로 끌고 들어오는 방법입니다.
작성하신 코드와 다른 점은 중간에 상태값이 만들어지지 않는 다는 점이 있지만 제 생각에 큰 차이는 없습니다.
두 방법 모두 괜찮아 보입니다.
```suggestion
            .filter(name -> name.length() > MAX_NAME_LENGTH)
            .findAny()
            .ifPresent(overLength -> {
                throw new RuntimeException("이름은 5자 이하여야 합니다.");
            });
```